metrics/report/main.go

```diff
@@ -0,0 +1,88 @@
+/// Copyight (c) 2017 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"flag"
+	"github.com/BurntSushi/toml"
+	"log"
+	"os/exec"
+	"strings"
+)
+
+// Configuration stores all necessary
+// information for checktool execution
+// and sent output as a report by email.
+type Configuration struct {
+	Mail MailConf
+	Ck   Checktool `toml:"checktool"`
+}
+
+// MailConf stores the authentication
+// configuration for sending the report
+// by email. This information is parsed
+// from TOML input file.
+type MailConf struct {
+	Smtp    string   // SMTP server address
+	Port    string   // SMTP port
+	User    string   // SMTP user
+	Pass    string   // SMTP password
+	Id      string   // SMTP indetity
+	From    string   // Sender
+	Subject string   // Email msg subject
+	To      []string // Receipts
+	Cc      []string // Msg copy
+}
+
+// Checktool stores the arguments
+// for checktool execution
+type Checktool struct {
+	Basefile   string
+	Metricsdir string
+}
+
+func main() {
+
+	var confFile string
+	var conf Configuration
+	var cmd string
+	var outRows []string
+
+	// Checktool conf vars
+	var basefile string
+	var metricsdir string
+
+	flag.StringVar(&confFile, "f", "", "Configuration file")
+	flag.Parse()
+
+	// Parsing TOML configuration file
+	if _, err := toml.DecodeFile(confFile, &conf); err != nil {
+		log.Fatal(err)
+	}
+
+	// Set checktool configuration
+	cmd = "checktool "
+	basefile = "--basefile " + conf.Ck.Basefile
+	metricsdir = "--metricsdir " + conf.Ck.Metricsdir
+
+	// Checktool execution
+	out, err := exec.Command(cmd, basefile, metricsdir).Output()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	outRows = strings.Split(string(out), "\n")
```

Comment:
Just a heads up that `checkmetrics` just changed its output to a more formatted table using https://github.com/olekukonko/tablewriter (see
https://github.com/clearcontainers/tests/pull/98/files#diff-1b7dc2c4625a82d201f9557ff3035f82R94 for example output) - so we may need to check, or update later, to accommodate that.
As mentioned before, if we need a more parseable or email friendly format (although that table is OK I think), then we can add something like a `--json` option to `checkmetrics`.